src/main/java/io/cdap/plugin/google/sink/GoogleDriveSink.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.google.sink;
+
+import com.google.api.services.drive.model.File;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.batch.Output;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.dataset.lib.KeyValue;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchSink;
+import io.cdap.cdap.etl.api.batch.BatchSinkContext;
+import io.cdap.plugin.google.common.FileFromFolder;
+import io.cdap.plugin.google.common.GoogleDriveBaseConfig;
+import org.apache.commons.lang3.RandomStringUtils;
+
+/**
+ * Batch sink to writing multiple files to Google Drive directory.
+ */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name("GoogleDrive")
+@Description("Sink plugin to save files from the pipeline to Google Drive directory.")
+public class GoogleDriveSink extends BatchSink<StructuredRecord, Void, Void> {
+
+  private static final String NAME_FIELD_NAME = "name";
+
+  private final GoogleDriveSinkConfig config;
+  private final GoogleDriveSinkClient sinkClient;
+
+  public GoogleDriveSink(GoogleDriveSinkConfig config) {
+    this.sinkClient = new GoogleDriveSinkClient(config);
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    config.validate(failureCollector);
+    validateSchema(failureCollector, pipelineConfigurer.getStageConfigurer().getInputSchema());
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext batchSinkContext) {
+    batchSinkContext.addOutput(Output.of(config.referenceName, new GoogleDriveOutputFormatProvider()));
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<KeyValue<Void, Void>> emitter) throws Exception {
+    sinkClient.createFile(convertFromMessage(input));
```

Comment:
Usually we use recordWriter to do this.
`convertFromMessage(input)` is done in trasform
and `sinkClient.createFile` is done in `RecordWriter<NullWritable, FileFromFolder>`